### PR TITLE
[stdlib] Conditional gardening [NFC]

### PR DESCRIPTION
**Pull request by moiseev**

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Review by DougGregor — COMMENTED** [review]
There are three different indentations for the `where` in a where clause in this PR.

**Comment by moiseev** [comment]
@DougGregor yes.

File: stdlib/public/core/Indices.swift [hunk 1]
```diff
@@ -130,5 +130,12 @@ extension Collection where Indices == DefaultIndices<Self> {
   }
 }
 
-public typealias DefaultBidirectionalIndices<T: BidirectionalCollection> = DefaultIndices<T>
-public typealias DefaultRandomAccessIndices<T: RandomAccessCollection> = DefaultIndices<T>
+@available(*, deprecated, renamed: "DefaultIndices")
+public typealias DefaultBidirectionalIndices<
+  T : BidirectionalCollection
+> = DefaultIndices<T>
```
Comment (airspeedswift) [inline stdlib/public/core/Indices.swift]: I feel like you're just 80-col-trolling me now...

**Review by airspeedswift — APPROVED** [review]
LGTM

File: stdlib/public/core/Indices.swift (hunk 1 above)
Comment (moiseev) [reply stdlib/public/core/Indices.swift]: Reverted. It doesn't make much sense here.

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

**Comment by moiseev** [comment]
For the record. The failure is:

```
--
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:193:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:193:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:414:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomRangeReplaceableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:414:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomRangeReplaceableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:599:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:599:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:832:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRangeReplaceableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:832:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRangeReplaceableRandomAccessButt>
                      ^

--
```

**Review by DougGregor — APPROVED** [review]
LGTM

**Comment by moiseev** [comment]
@swift-ci Please test and merge

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by airspeedswift** [comment]
@swift-ci please test macOS platform

**Pull request merged**